src/nlsolve/nlsolve.jl

```diff
@@ -3,15 +3,17 @@
 
 Solve
 ```math
-dt⋅f(tmp + γ⋅z, p, t + c⋅dt) = z
+dt⋅f(innertmp + γ⋅z, p, t + c⋅dt) + outertmp = z
 ```
 where `dt` is the step size and `γ` and `c` are constants, and return the solution `z`.
 """
 function nlsolve!(nlsolver::AbstractNLSolver, integrator, cache=nothing, repeat_step=false)
   @label REDO
   if isnewton(nlsolver)
-    cache isa Nothing && throw(ArgumentError("cache is not passed to `nlsolve!` when using NLNewton"))
-    update_W!(nlsolver, integrator, cache, nlsolver.γ*integrator.dt, repeat_step)
+    cache === nothing && throw(ArgumentError("cache is not passed to `nlsolve!` when using NLNewton"))
+    γW = nlsolver.γ*integrator.dt
+    isodemultistep(unwrap_alg(integrator, true)) && (γW = γW / nlsolver.α)
+    update_W!(nlsolver, integrator, cache, γW, repeat_step)
```

Comment:
Maybe one could even use a proper `if` - `else` statement to make it more obvious what's happening here. It feels like the difference between both methods is a bit hidden, and maybe an explicit
```suggestion
    if isodemultistep(unwrap_alg(integrator, true))
        update_W!(nlsolver, integrator, cache, γW / nlsolver.α, repeat_step)
    else
        update_W!(nlsolver, integrator, cache, γW, repeat_step)
    end
```
would make it clearer.